iaso/api/profiles.py

```diff
@@ -15,19 +16,50 @@
 from django.utils.http import urlsafe_base64_encode
 from django.utils.translation import gettext as _
 from rest_framework import viewsets, permissions, status
-from rest_framework.exceptions import ValidationError
+from rest_framework.exceptions import PermissionDenied, ValidationError
 from rest_framework.response import Response
 
 from iaso.models import Profile, OrgUnit, UserRole, Project
 from hat.menupermissions import models as permission
+from hat.menupermissions.models import CustomPermissionSupport
+
+PK_ME = "me"
 
 
 class HasProfilePermission(permissions.BasePermission):
     def has_permission(self, request, view):
-        if view.action in ("retrieve", "partial_update") and view.kwargs.get("pk") == "me":
+        if view.action in ("retrieve", "partial_update") and view.kwargs.get("pk") == PK_ME:
+            return True
+        if request.user.has_perm(permission.USERS_ADMIN):
             return True
-        if (not request.user.has_perm(permission.USERS_ADMIN)) and request.method != "GET":
-            return False
+        if request.user.has_perm(permission.USERS_MANAGED):
+            return self.has_permission_over_user(request, view)
+
+        return request.method == "GET"
+
+    # We could `return False` instead of raising exceptions,
+    # but it's better to be explicit about why the permission was denied.
+    @staticmethod
+    def has_permission_over_user(request, view):
+        pk = view.kwargs.get("pk")
+        if not pk:
+            raise PermissionDenied(f"User with '{permission.USERS_MANAGED}' cannot create users.")
+
+        if pk == request.user.id:
+            raise PermissionDenied(f"User with '{permission.USERS_MANAGED}' cannot edit their own permissions.")
+
+        org_units = OrgUnit.objects.hierarchy(request.user.iaso_profile.org_units.all()).values_list("id", flat=True)
```

Comment:
I'm questioning this. A user that can modify the permissions of everybody without creating one could be an interesting option too and would maybe match better what we do with org units limitation (basically, you're more powerful if you have no org units assigned than if you have some)